packages/lesswrong/components/ea-forum/TargetedJobAdSection.tsx

```diff
@@ -73,53 +83,103 @@ const TargetedJobAdSection = () => {
   
   // select a job ad to show to the current user
   useMemo(() => {
-    if (!currentUser || userJobAdsLoading || coreTagReadsLoading || activeJob) return
-    
-    // user's relevant interests from EAG, such as "software engineering"
-    // TODO: add this back in once we have the data
-    // const userEAGInterests = union(currentUser.experiencedIn, currentUser.interestedIn)
+    if (!currentUser || userJobAdsLoading || userEAGDetailsLoading || coreTagReadsLoading || activeJob) return
+  
     const ads = userJobAds ?? []
     
     for (let jobName in JOB_AD_DATA) {
+      const jobAd = JOB_AD_DATA[jobName]
       // skip any jobs where the deadline to apply has passed
-      const deadline = JOB_AD_DATA[jobName].deadline
+      const deadline = jobAd.deadline
       if (deadline && moment().isAfter(deadline, 'day')) {
         continue
       }
 
-      const jobAdState = ads.find(ad => ad.jobName === jobName)?.adState
-      // check if the ad fits the user's interests -
-      // currently based on whether they have subscribed to all the topics relevant to the job ad
-      const subscribedTagIds = JOB_AD_DATA[jobName].subscribedTagIds
+      const userJobAdState = ads.find(ad => ad.jobName === jobName)?.adState
+      const userEAGData = userEAGDetails?.[0]
```

Comment:
I think it would be a bit better to do all this filtering logic in a resolver on the server, to reduce the amount of data that has to be returned. Maybe the country code makes that a bit tricky to do though. Anyway I don't think it's worth changing for this PR, especially seeing as we only have a small number of jobs, but could be worth coming back and refactoring

Reply:
Seems reasonable, we do consider more sources of data now. It also depends on which of these conditions we keep (I could see us removing the reading one, in which case that may not be worth it anymore)